diff --git a/dist/module.cjs b/dist/module.cjs
index 7806b2b99c4d10fec58361920b408bd714afc0d8..74462dea500c4ef737b14b707e72fb3f305186c4 100644
--- a/dist/module.cjs
+++ b/dist/module.cjs
@@ -292,6 +292,7 @@ async function getLocalDatabase(database, { connector, nativeSqlite } = {}) {
   };
 }
 function findBestSqliteAdapter(opts) {
+  return "db0/connectors/sqlite3"
   if (process.versions.bun) {
     return "db0/connectors/bun-sqlite";
   }
diff --git a/dist/module.mjs b/dist/module.mjs
index 35fda6b0359b88fa90a117a61895bc09fcf7329c..f5d3f280f487dd940d71e4f224eebd8099bc13e7 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -272,6 +272,7 @@ async function getLocalDatabase(database, { connector, nativeSqlite } = {}) {
   };
 }
 function findBestSqliteAdapter(opts) {
+  return "db0/connectors/sqlite3"
   if (process.versions.bun) {
     return "db0/connectors/bun-sqlite";
   }
